lxd/storage/utils.go

```diff
@@ -706,6 +706,12 @@ func VolumeUsedByInstances(s *state.State, poolName string, projectName string,
 	volumeNameWithType := fmt.Sprintf("%s/%s", vol.Type, vol.Name)
 
 	return s.Cluster.InstanceList(func(inst db.Instance, p api.Project, profiles []api.Profile) error {
+		//If the volume has a specific cluster member which is different than the instance then skip as
+		// instance cannot be using this volume.
+		if vol.Location != "" && inst.Node != vol.Location {
```

Comment:
@stgraber this is the part we were discussing where we exclude instances from being considered as users of a volume if the volume's location is not the same as the instance (or if the volume is "global" and doesn't have a location set).

This logic is based on the behaviour of the function that looks up a storage volume:

https://github.com/lxc/lxd/blob/269588e3f7586eb9438a9268cee9d238150afc86/lxd/db/storage_volumes.go#L291-L296

Which explicitly doesn't populate the Location field for remote storage drivers.